### DISCUSSION TITLE
Add 7th Sea 2nd Edition dice roller

### DISCUSSION THREAD

**Discussion by JSpring3** [LokiMidgard/Dice-Tool#2] [Ideas]
I would be cool if this did 7th Sea 2nd Edition dice.
I looked into this before but didn't know how to solve it.

https://stackoverflow.com/questions/69416369/how-to-make-subsets-of-numbers-equaling-10-or-more-from-a-master-set

Here is the information from the rulebook.

"After you roll, use your dice to create sets of 10. Each
set of 10 you create is a Raise. Sometimes, you may
use dice that add up to more than 10. That’s okay; it’s
still a Raise. But if you don’t have enough points to
make a 10, you can’t use those dice for a Raise.
I roll 10, 7, 5, 5, 2 and 2. I can make these sets
of 10:
10 (1 Raise),
5+5 = 10 (1 Raise),
7+2+2 = 11 (1 Raise)
Another example: I roll 8, 4, 3, 3, 2, 1, 1.
8+1+1 = 10 (1 Raise),
4+3+3 = 10 (1 Raise)
I have a 2 left over. I can’t make a Raise out
of that.
Another Example: I roll 10, 9, 9, 7.
10 (1 Raise),
9+7 = 16 (1 Raise)
I have a 9 left over. I can’t make a Raise out
of that."

**Comment by LokiMidgard** [comment]
This will propably not be so easy.

It's some time I've written this, but if I remember correctly, this project currently does not support sets or lists. Without that it will be a little hard to make something you need. Not impossible but you will need much more code.

I think you could try to use the "older" version which was not as performand as this one but could use "normal" c# code and just tries every combiniation of roled dices. I think its under the branch `version-1`

The other problem you have is the actual algorithm. Besides burtforcing every combination, you could search for algorithms that could help you. With a quick search I havn't found a solution.

I'm not sure how trained your in algorithm desgin. But I guess some keywords you can try to search for are `partion` and `subsets`. (if you haven't done it already)

Maybe you find a hint in [Multiple subset sum](https://en.wikipedia.org/wiki/Multiple_subset_sum) Wiki artikle. I think your problem could be `Min-sum MSSP` instead of `Max-sum MSSP`. But I haven't found anything in thet direction yet.
